internal/controllers/clusterextension_controller.go

```diff
@@ -387,28 +387,29 @@ func (r *ClusterExtensionReconciler) generateExpectedBundleDeployment(o ocv1alph
 	// unstructured ensures that the patch contains only what is specified. Using unstructured like this is basically
 	// identical to "kubectl apply -f"
 
+	spec := map[string]interface{}{
+		// TODO: Don't assume plain provisioner
+		"provisionerClassName": bundleProvisioner,
+		"source": map[string]interface{}{
+			// TODO: Don't assume image type
+			"type": string(rukpakv1alpha2.SourceTypeImage),
+			"image": map[string]interface{}{
+				"ref": bundlePath,
+			},
+		},
+	}
+
+	if len(o.Spec.WatchNamespaces) > 0 {
+		spec["watchNamespaces"] = o.Spec.WatchNamespaces
+	}
```

Comment:
Is it worth adding test coverage for this?

Reply:
I've added an unit test for the method for now. An test to support this particular feature is already in rukpak. I'll create an issue to follow up and add an e2e test with a catalog for this.